File: algorithm/DFS.cpp
#include "jsoncppheader.h"
#include <string>
DirectedGraphTracer tracer = DirectedGraphTracer();
LogTracer logger = LogTracer();
int G[11][11] = {
	{0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0},
    {0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
};
void DFS(int node,int parent) { 
	if(parent != -1) tracer.visit(node, parent).wait();
	for (int i = 0; i < 11; i++) {
		if (G[node][i]) { 
			DFS(i, node); 
		}
	}
}
int main()
{
	tracer.setTreeData(G[0], 11, 11, 0);
	tracer.attach(logger.id);
	DFS(0, -1);
}

File: algorithm/python/graph/dfs.py
# -*-encoding:utf-8-*-
from tracer import *
logger = LogTracer()
tracer = DirectedGraghTracer().attach(logger.getID())
G = [ # G[i][j] indicates whether the path from the i-th node to the j-th node exists or not
    [0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
]
tracer.setData(G)


def dfs(u, pre):
    tracer.visit(u, pre).wait()
    #print('vis', u)
    for i in range(len(G[u])):
        if G[u][i]:
            dfs(i, u)


if __name__ == '__main__':
    dfs(0, -1)
    tracerlist.info()